src/pkg/task/model.go

```diff
@@ -52,6 +55,16 @@ type Execution struct {
 	EndTime    time.Time              `json:"end_time"`
 }
 
+// IsOnGoing returns true when the execution is pending/running
+func (exec *Execution) IsOnGoing() bool {
```

Comment:
`Execution` has no `pending` status, so you can just determine whether an execution is ongoing by comparing the status with `running`